File: WebClient SDK/src/default/abTest/abExternalFileExtension.ts
import { AbExternalFile } from './abExternalFile';
import { FileExtension, IFileExtensionContext } from 'tessa/ui/files';
import { UIContext } from 'tessa/ui';
import { ICardModel } from 'tessa/ui/cards';
import { extension } from '@tessa/application';

@extension()
export class AbExternalFileExtension extends FileExtension {
  public openingMenu(context: IFileExtensionContext): void {
    // Проверяем тип карточки
    const editor = UIContext.current.cardEditor;
    let model: ICardModel;
    if (editor == null || !(model = editor.cardModel!) || model.cardType.name !== 'AbCar') {
      return;
    }

    // Отключаем пункт "Копировать ссылку"
    const copyLinkAction = context.actions.find(p => p.name === 'CopyLink');
    if (context.file.model instanceof AbExternalFile && copyLinkAction) {
      copyLinkAction.isCollapsed = true;
    }
  }
}

File: Extensions/Tessa.Extensions.Default.Client/AbTest/AbExternalFileExtension.cs
using System.Threading.Tasks;
using Tessa.Extensions.Default.Shared.AbTest;
using Tessa.UI;
using Tessa.UI.Files;

namespace Tessa.Extensions.Default.Client.AbTest
{
    /// <summary>
    /// Расширение, инициализирующее контекстное меню для файлов значениями по умолчанию.
    /// </summary>
    public sealed class AbExternalFileExtension :
        FileExtension
    {
        #region Base Overrides

        public override async Task OpeningMenu(IFileExtensionContext context)
        {
            // Проверяем тип карточки
            if (UIContext.Current.CardEditor.CardModel is not { } model
                || model.CardType.ID != AbCardTypes.AbCarTypeID)
            {
                return;
            }

            // Отключаем пункт "Копировать ссылку"
            if (context.File is AbExternalFile
                && context.Actions.TryGet(FileMenuActionNames.CopyLink) is { } copyLinkAction)
            {
                copyLinkAction.IsEnabled = false;
                copyLinkAction.IsCollapsed = true;
            }
        }

        #endregion
    }
}
